libs/constrained-generators/src/Constrained/Spec/Map.hs

```diff
@@ -295,9 +297,13 @@ instance BaseUniverse fn => Functions (MapFn fn) fn where
           | NilCtx HOLE <- ctx
           , Evidence <- prerequisites @fn @(Map k v) ->
               case spec of
-                MemberSpec [s] ->
-                  typeSpec $
-                    MapSpec Nothing s [] (equalSpec $ sizeOf s) TrueSpec NoFold
+                MemberSpec es
+                  | NE.length es == 1
+                  , let s = NE.head es
```

Comment:
It should be possible to do this by pattern matching instead. That goes for a number of places.

Reply:
Ah yes, I forgot about this strange constructor for NonEmpty

a :| [a] infixr 5

Reply:
So what about this as a more general solution?
 
MemberSpec es ->  typeSpec $ MapSpec Nothing s [] (MemberSpec (fmap sizeOf es) TrueSpec NoFold

for this more specific one

MemberSpec (s :| []) ->
                  typeSpec $
                    MapSpec Nothing s [] (equalSpec $ sizeOf s)

Reply:
OK the general solution doesn't make sense.